_posts/2016-12-14-four-tips-sass-variables.markdown

```diff
@@ -0,0 +1,237 @@
+
+### Know the Different Variable Types
+
+There's Sass variable types that Javascript writers may be familiar with. The variable type directly affects how the value is called and how complex it is, so this a good starting point for learning more.
+
+#### Basic Variables
+
+Basic variables are the simplest: singular values for a CSS property. They can range from basic colors (`#FF0000`) or lengths (`1em`), to ones that hold a few words, like `1px solid black` for borders. Their main purpose is substituting a value for a CSS property.
+
+Here's a simple example:
+
+{% highlight sass %}
+$light-color: #111;
+
+.element { background-color: $light-color; }
+{% endhighlight %}
+
+#### List variables
+
+List variables are like Javascript arrays: they contain a list of values the code will usually loop through. If there's several colors that only work in a group, a developer can put them in a list for more efficient use.
+
+Accessing a list's values needs either an *@each loop* or *getting a value at its index.* Here's an example of each one:
+
+{% highlight sass %}
+// This is the list
+$element-colors: #F00, #0F0, #00F;
+
+// This is the @each loop. In each iteration, $color is
+// the name we chose for the current value
+@each $color in $element-colors {
+    $i: index($element-colors, $color); // This returns the list index of the color
+
+    // This makes use of the index and value,
+    // so the loop creates different selectors for each value
+    .element:nth-of-type(#{$i}) { background-color: $color; }
+}
+
+// This is calling a color by the index alone
+.another-element {
+    
+    background-color: nth($element-colors, 1); // returns #F00
+}
+{% endhighlight %}
+
+You'll notice the start of a Sass list's index is 1, not 0. This often throws off many people who've only used Javascript arrays before.
+
+Also note the `@each` loop uses the `$i` variable in it's selector. Using that, or even the `$color` variable, is important. Otherwise it'd create the same `.element` selector three times, and only the last one would work. Remember to use these variables for selectors in `@each` loop for this reason, and in that context reference them like this:`#{$i}`.
+
+#### Map Variables
+
+Map variables are the most complex. They're like Javascript objects since **they store object-value pairs,** and work for more complex data like this:
+
+{% highlight sass %}
+$color-map: (
+    color1 : #F00,
+    color2 : #0F0,
+    color3 : #00F
+);
+{% endhighlight %}
+
+Similar to lists, `@each` loops are used to access the values. Developers can also access values by their specific keys. So for the different colors in this map, we could do this:
+
+{% highlight sass %}
+// We assign names to the properties and values in the map
+@each $name, $color in $color-map {
+    
+    // We can use them both for more detailed selectors in the loop
+    .#{$name}-background { background-color: $color; }
```

Comment:
So this would create `.color1-background { background-color: #F00; }`, etc, right? (just for my own understanding)